clients/client/src/main/java/org/projectnessie/client/StreamingUtil.java

```diff
@@ -43,16 +48,30 @@ private StreamingUtil() {
    * Default implementation to return a stream of references, functionally equivalent to calling
    * {@link NessieApiV1#getAllReferences()} with manual paging.
    *
+   * @param api The {@link NessieApiV1} to use
+   * @param builderCustomizer a Consumer of the {@link GetAllReferencesBuilder} that is passed to
+   *     the caller of this method. It allows the caller to customize the GetAllReferencesBuilder
+   *     with additional parameters (e.g.: filter).
+   * @param maxRecords - a maximum number of records in the stream. If it is {@code
+   *     Optional.empty()} the stream will be unbounded.
+   * @param filter - a filter expression to use when fetching a stream of references.
    * @return stream of {@link Reference} objects
    */
   public static Stream<Reference> getAllReferencesStream(
-      @NotNull NessieApiV1 api, OptionalInt maxRecords, @Nullable String filter)
+      @NotNull NessieApiV1 api,
+      @NotNull Consumer<GetAllReferencesBuilder> builderCustomizer,
```

Comment:
Should we allow the consumer to be `null` in case you don't need any customization?

Reply:
I am not sure if this is a good pattern to pass null as a `Consumer`.
The caller can pass a no_op consumer if he does not wish to alter the behavior.
 I was thinking about creating:
`public static final Consumer< GetAllReferencesBuilder> NO_OP_GET_ALL_REFERENCES = ...`
but we need to create it for all 4 types, so it is becoming verbose. 
The other solution would be to create an overloaded method without the:
`@NotNull Consumer<GetAllReferencesBuilder> builderCustomizer` parameter, and let the `StreamUtil` pass `(c) -{}` instead of the caller. I would favor the latter solution. Wdyt?

Reply:
Yea - let's just keep it `@NotNull`. No need for constants (`x -> {}` is short enough).